github/resource_github_team.go

```diff
@@ -243,6 +243,29 @@ func resourceGithubTeamDelete(d *schema.ResourceData, meta interface{}) error {
 
 	log.Printf("[DEBUG] Deleting team: %s", d.Id())
 	_, err = client.Teams.DeleteTeamByID(ctx, orgId, id)
+	/*
+		When deleting a team and it failed, we need to check if it has already been deleted meanwhile.
+		This could be the case when deleting nested teams via Terraform by looping through a module
+		or resource and the parent team might have been deleted already. If the parent team had
+		been deleted already (via parallel runs), the child team is also already gone (deleted by
+		GitHub automatically).
+		So we're checking if it still exists and if not, simply remove it from TF state.
+	*/
+	if err != nil {
+		// Fetch the team in order to see if it exists or not (http 404)
+		_, _, err = client.Teams.GetTeamByID(ctx, orgId, id)
+		if err != nil {
+			if ghErr, ok := err.(*github.ErrorResponse); ok {
+				if ghErr.Response.StatusCode == http.StatusNotFound {
+					// If team we failed to delete does not exist, remove it from TF state.
+					log.Printf("[WARN] Removing team: %s from state because it no longer exists",
```

Comment:
The following warning does not show up with `TF_LOG=WARN`. It only shows up in `TF_LOG=DEBUG`. Not sure if this is a general issue with this provider or just with me doing something wrong in the code:

Reply:
@jcudit the only thing I'd need some help on is the logging level as the above is only showing up in `TG_LOG=DEBUG`. Example line below:
```
2022-01-22T10:47:11.813Z [DEBUG] provider.terraform-provider-github_v4.19.1-fl.4: 2022/01/22 10:47:11 [WARN] Removing team: 5592187 from state because it no longer exists
```
However, when looking at a valid `WARN` message, it looks like this:
```
2022-01-22T10:47:04.978Z [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/flaconi/github\"]" changed the config value, but that value is unused
```

Any idea on what I'm doing wrong here?

Reply:
I am stumped.  This looks very similar to what I'm seeing in [`hashicorp/terraform`](https://github.com/hashicorp/terraform/search?p=2&q=warn), so 😕 why it is misbehaving for you.

Does `TF_LOG=JSON` do anything different? Perhaps `TG_LOG` was used in error?